src/update.c

```diff
@@ -675,6 +682,15 @@ enum swupd_code update_main(int argc, char **argv)
 		ret = check_update();
 	} else {
 		ret = execute_update();
+		if (cmdline_option_3rd_party) {
+			if (ret == SWUPD_OK) {
+				progress_finish_steps(ret);
+				info("\nUpdating content from 3rd-party repositories\n");
+				ret = third_party_execute_update(steps_in_3rd_party_update);
```

Comment:
I keep with my comment. Why do you need to move the information on how many steps are needed on 3rd_party update to update.

Reply:
I'm sorry I don't understand your question, `third_party_run_operation_multirepo()` requires the number of steps of the operation to be run as a parameter.

Reply:
Yes. But you are not calling `third_party_run_operation_multirepo` here. You are calling `third_party_execute_update`. Can't you change `third_party_execute_update` to have no parameters and check the number of steps there instead of setting it here?

Reply:
ah I see what you mean now. Yeah I thought of that at first, but we need to know if the user used the `--download` flag in order for us to calculate the number of steps accurately. I could send that as parameter to third_party_execute_update instead of sending the calculated number of steps.